doc/reference/networking/net_config.rst

```diff
@@ -1,11 +1,69 @@
 .. _net_config_interface:
 
-Configuration Library
-######################
+Network Configuration Library
+#############################
 
 Overview
 ********
 
+The network configuration library can be used to setup the device in
+semi-automatic way when it is booted. The user just needs to setup
+various Kconfig options and the config library tries to setup the
+device accordingly.
+
+The following Kconfig options affect how configuration library will
+setup the system.
+
+.. csv-table:: Kconfig options for network configuration library
+   :header: "Option name", "Description"
+   :widths: auto
+
+   "CONFIG_NET_CONFIG_SETTINGS", "This option controls whether the network
```

Comment:
It would be good to have all these references to Kconfig options as links to their documentation...
```suggestion
   ":option:`CONFIG_NET_CONFIG_SETTINGS`", "This option controls whether the network
```

Reply:
Very true, I missed this one.